### PR TITLE
cli integration of replicaset and moveto config file

### LINKED ISSUE #11: Configuration system
- Initial version should be able to be initialized with a config file

### PR DESCRIPTION
**Pull request by drubin**
Adds a basic example usage and setup with a config file.

We can iterate on the config file format for me the biggest issue with the current setup is 

1. Validation (we don't validate configuration before using it), lead to a few issues
2. I don't love the name `replicaSet.replicaSetName` I think it should be `replicaSet.name` this reads better in the config and in the code. 

closes #11

### CODE REVIEW

File: examples/replication/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+* node >= 10
+* npm
+
+
+### Steps
+
+1. Start instances `docker-compose up`
+2. Run hydra `hydra replication -f hydra.yaml`
+
+You now have a 3 node mongo replica set!
```
Comment (pelzerim) [inline examples/replication/README.md:18]: !!

File: examples/replication/hydra.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Define a replica set
+# to support multiple replica sets use multiple config  files
```
Comment (pelzerim) [inline examples/replication/hydra.yaml:2]:
```suggestion
```

Lets rather make a seperate example (not in this mr though)

File: examples/replication/hydra.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+replicaSet:
+  # Replica set name
+  replicaSetName: mongo-rs
+  # The servers that should be apart of this replica set
```
Comment (pelzerim) [inline examples/replication/hydra.yaml:6]:
```suggestion
  # The servers that should be part of this replica set
```

File: examples/replication/hydra.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  # host: The host mongo internally uses to name and connect to these
+  # uri: the  uri that  hydra uses  to connect and configure (can be a mapping/port-forwarding/public uri)
```
Comment (pelzerim) [inline examples/replication/hydra.yaml:8]:
```suggestion
  # uri: the URI that  hydra uses  to connect and configure (can be a mapping/port-forwarding/public URI)
```

File: examples/replication/README.md (hunk 1 above)
Comment (drubin) [reply examples/replication/README.md:18]: Must I do something here?

File: examples/replication/hydra.yaml (hunk 2 above)
Comment (drubin) [reply examples/replication/hydra.yaml:2]: Sure later.

File: examples/replication/docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    image: "mongo:${MONGO_VERSION:-4.2}"
+    command:
+    - '--replSet'
+    - 'mongo-rs'
+    - '--port'
+    - '27001'
+    ports:
+    - "27001:27001"
+  mongo-rs-2:
+    image: "mongo:${MONGO_VERSION:- 4.2}"
```
Comment (pelzerim) [inline examples/replication/docker-compose.yaml:22]:
```suggestion
    image: "mongo:${MONGO_VERSION:-4.2}"
```

File: examples/replication/README.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+* Docker
+* Docker compose
+* node >= 10
+* npm
+
+### Steps
+
+1. Start instances `docker-compose up`
+2. Run hydra `hydra replication -f hydra.yaml`
```
Comment (pelzerim) [inline examples/replication/README.md:15]: Missing install step

File: lib/cli/replication.js [hunk 7]
```diff
@@ -1,13 +1,25 @@
...
+  const state = new ReplicaSetState(replicaSet)
+  await state.apply()
+
+  await state.disconnect()
+  if (state.replicaSetStatus === REPLICA_SET_STATUS.HEALTHY) {
+    logger.info('+ ReplicaSet status is healthy')
+  } else {
+    logger.warn('- ReplicaSet status not healthy')
+    // this will cause an exit code
+    throw new Error('replica set not healthy')
```
Comment (pelzerim) [inline lib/cli/replication.js:23]:
```suggestion
    throw new Error(`ReplicaSet not healthy. Has state ${state.replicaSetStatus}`)
```

**Pull request merged**